### PR TITLE
New disk structure for collect disk details

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Signed-off-by: GowthamShanmugam <gshanmug@redhat.com>

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
@r0h4n @brainfunked @anivargi @shtripat @nnDarshan @nthomas-redhat please review

File: tendrl/node_agent/node_sync/disk_sync.py [hunk 1]
```diff
@@ -2,212 +2,308 @@
...
+                disks_map[devlist['hardware_id']] = devlist["disk_id"]
+            else:
+                Event(
+                    Message(
+                            priority="error",
+                            publisher=NS.publisher_id,
+                            payload={"message": err}
+                        )
+                )
+    return disks, disks_map, err
```
Comment (r0h4n) [inline tendrl/node_agent/node_sync/disk_sync.py]: raise ValueError if there is any "err" , catch the ValueError in calling function

**Comment by r0h4n** [comment]
Verified?

**Comment by brainfunked** [comment]
A change in the structure based on https://github.com/Tendrl/commons/pull/492#issuecomment-302306277 could be beneficial long term.

**Pull request merged**